### PR TITLE
JSHint Support

### PR DESCRIPTION
**Pull request by GordonSmith**

### CODE REVIEW

**Comment by GordonSmith** [comment]
@anmoljagetia, @mlzummo, @jbrundage  Please Review

File: src/amchart/Candle.js [hunk 1]
```diff
@@ -13,13 +13,12 @@
...
-        this._categoryField;
-        this._lowField;
-        this._highField;
-    };
-    
+        this._openField = undefined;
+        this._closeField = undefined;
+        this._categoryField = undefined;
+        this._lowField = undefined;
+        this._highField = undefined;
```
Comment (jbrundage) [inline src/amchart/Candle.js]: Is it necessary to initialize variables in this way?

Comment (GordonSmith) [inline src/amchart/Candle.js]: ...no... But the previous code does nothing and I am assuming the only reason they are there is to "self document" - IMO if they are published params, then they should be removed altogether (but that should be in a separate PR)?

(Actually now that you point it out - if they are published params, then they are now just wrong  and really _need_ to be be deleted).

File: src/amchart/Pie.js [hunk 2]
```diff
@@ -93,7 +92,7 @@
         
         this.pieAlpha().forEach(function(d,i) {
             if (typeof(this._chart.chartData[i])==='undefined') { 
-                this._chart.chartData[i] = {} 
+                this._chart.chartData[i] = {};
```
Comment (jbrundage) [inline src/amchart/Pie.js]: When should/shouldn't a semi-colon be used?

Comment (GordonSmith) [inline src/amchart/Pie.js]: In this specific case you need it at the end of every assignment (once this PR is pulled the answer will be "whenever JSHint tells you"<g>).

I think the JS rule is that it is needed to separate JS statements, which is why it is not needed  at the end of code blocks (function xxx() {}, if {} and switch {}) or for the the last statement within a code block.  FWIW JSHint will insist you have them for the last statement in a code block (which is good with me!).

File: src/amchart/Candle.js (hunk 1 above)
Comment (mlzummo) [inline src/amchart/Candle.js]: actually we want to init those are empty arrays as those are private vars (not publish params)

Comment (GordonSmith) [inline src/amchart/Candle.js]: If you want to init them - then open a diff PR - the goal of this PR is to be JSHint compliant with no logic changes...

**Comment by mlzummo** [comment]
looks good to me
